packages/back-end/src/integrations/SqlIntegration.ts

```diff
@@ -3697,7 +3860,11 @@ export default abstract class SqlIntegration
   ${"ignoreNulls" in metric && metric.ignoreNulls ? `WHERE m.value != 0` : ""}
   GROUP BY
     m.variation
-    , ${cumulativeDate ? `${this.formatDate("m.day")}` : "m.dimension"}
+    ${
+      dimensionCols.length > 0
+        ? `, ${dimensionCols.map((c) => `m.${c.alias}`).join(", ")}`
+        : ""
+    }
```

Comment:
Nit: this is a simpler way to do this without having the check the length everywhere.

```ts
${dimensionCols.map((c) => `, m.${c.alias}`).join("")}
```